Atomic/syscontainers.py

```diff
@@ -210,6 +210,40 @@ def _find_rpm(tmp_dir):
         return rpm_file
 
     def install(self, image, name):
+        """
+        External container install logic.
+
+        :param image: The name of the image
+        :type image: str
+        :param name: The name of the checkout
+        :type name: str
+        """
+        # If we have oneshot
+        if self.args.oneshot and (self.args.system or self.args.user):
+            try:
+                # Force system_package to no if this is a oneshot
+                self.args.system_package = 'no'
+                # Use the internal _install method to install the image
+                install_result = self._install(image, name)
+
+                # Get the start command for the system container
+                (start_command, _) = self._generate_systemd_startstop_directives(name)
+                # Move to the proper directory to start the system container
+                os.chdir(os.path.join(ATOMIC_VAR, name))
+                # ... and run it. We use call() because the actual run may be expected to fail.
+                util.call(start_command)
+            finally:
+                # Uninstall the image post run
+                self.uninstall(name)
```

Comment:
when I firstly suggested this I forgot we had an `extract`.  Sorry :)

At some point we will need an OSTree temporary storage (also to fix mount) but for now I think it would be better if we can do something like:

```python
try:
    self.extract(img, tmpdir)
    generate_systemd_directives(...)
    os.chdir("$tmpdir/rootfs")
    util.call(...)
finally:
   shutil.rmtree(tmp)
```

`tmpdir` is better created under the OSTree repository so we are sure to use hard links and the checkout is faster.  The `_try_ostree_mount` function in mount.py does that (we will need to copy/move it to syscontainers).  Anyway, I'll have to fix this for the RPM stuff as well so feel free to ignore this part.

What do you think?

I see in your TODO list that there is to create the runc directives for --user.  That should be already done by generate_systemd_startstop_directives now to use bwrap-oci so there should not be anything to do for this.